pkg/cfn/builder/vpc.go

```diff
@@ -63,10 +66,22 @@ func getSubnetIPv6CIDRBlock(cidrPartitions int) *gfnt.Value {
 	return refSubnetSlices
 }
 
-func getSubnetIPv4CIDRBlock(cidrPartitions int) *gfnt.Value {
-	//TODO: should we be doing /19? Should we adjust for the partition size?
-	desiredMask := 19
-	refSubnetSlices := gfnt.MakeFnCIDR(gfnt.MakeFnGetAttString("VPC", "CidrBlock"), gfnt.NewInteger(cidrPartitions), gfnt.NewInteger(32-desiredMask))
+func getSubnetIPv4CIDRBlock(cidrPartitions int, cidr *ipnet.IPNet) *gfnt.Value {
+	desiredMask := 13
+	// We only calculate it if a custom cidr range was given
+	// otherwise the hardcoded one is fine for now. Don't take my word on that.
+	if cidr != nil {
+		// To calculate the desiredMask -> ip -> 192.168.0.0/20 cidrPartition -> 6
+		// 32-20 -> 12 -> 2^12 -> 4096 -> 4096/cidrPartitions -> ~682 -> log2(682) -> ~9.2 -> 9 (we always floor)!
+		// This should result in subnets with bit size of 23! Because 32 - 9 -> 23! This is, however, calculated by
+		// the cloudformation CIDR function. We just need to pass in 9.
+		ones, _ := cidr.Mask.Size()
+		remainingCIDRBit := 32 - ones
+		remainingIPs := math.Pow(2, float64(remainingCIDRBit))
+		numberOfIPsPerSubnet := remainingIPs / float64(cidrPartitions)
+		desiredMask = int(math.Floor(math.Log2(numberOfIPsPerSubnet)))
+	}
+	refSubnetSlices := gfnt.MakeFnCIDR(gfnt.MakeFnGetAttString("VPC", "CidrBlock"), gfnt.NewInteger(cidrPartitions), gfnt.NewInteger(desiredMask))
```

Comment:
Trying to understand this at 9am

![image](https://user-images.githubusercontent.com/19669095/157630523-644da4b5-fa9e-492d-abc7-162d20c37049.png)